src/Library/src/DataUnit.cpp

```diff
@@ -905,95 +905,7 @@ std::istream& dataunitio::synchronise(std::istream &stream) {
-
-  switch ((unsigned char)type) {
-  case 0x00: d.type = SEQUENCE_HEADER; break;
-  case 0x10: d.type = END_OF_SEQUENCE; break;
-  case 0x20: d.type = AUXILIARY_DATA;  break;
-  case 0x30: d.type = PADDING_DATA;    break;
-  case 0xC8: d.type = LD_PICTURE;      break;
-  case 0xE8: d.type = HQ_PICTURE;      break;
-  case 0xCC: d.type = LD_FRAGMENT;     break;
-  case 0xEC: d.type = HQ_FRAGMENT;     break;
-  default:
-    d.type = UNKNOWN_DATA_UNIT;
-    throw std::logic_error("Stream Error: Nonconformant data unit type.");
-  }
-
-  Bytes next_parse_offset(4);
-  Bytes prev_parse_offset(4);
-
-  stream >> next_parse_offset >> prev_parse_offset;
-
-  if ((unsigned long) next_parse_offset == 0) {
-    switch (d.type) {
-    // Nonconformant for these
-    case SEQUENCE_HEADER:
-    case AUXILIARY_DATA:
-    case PADDING_DATA:
-      throw std::logic_error("Stream Error: Next parse offset illegally set to zero.");
-      break;
-    // Conformant - No bits to read 
-    case END_OF_SEQUENCE:
-      d.strm.str(std::string());
-      break;
-    // Conformant - Unknown number of bits to read 
-    case LD_PICTURE:
-    case HQ_PICTURE:
-    case LD_FRAGMENT:
-    case HQ_FRAGMENT:
-    {
-      // Next_parse_offset artificially constructed by looking for next parse_info header
-      Bytes next_type(1);
-      Bytes next_next_parse_offset(4);
-      Bytes next_prev_parse_offset(4);
-      
-      std::streampos data_unit_position = stream.tellg();
-      std::streampos next_data_unit_position;
-
-      do{
-        // Find next parse_info header
-        stream >> dataunitio::synchronise;
-        next_data_unit_position = stream.tellg();
-        stream>>next_type>>next_next_parse_offset >> next_prev_parse_offset;
-      }
-      // Checks that future data unit is not spurious by verifying that the prev_parse_offset corresponds
-      // to the actual offset (10.6.2).
-      while((int)data_unit_position!=(int)next_data_unit_position-(int)next_prev_parse_offset+9);
-
-      // Rewind stream
-      stream.seekg(data_unit_position);
-      
-      int bufsize = ((unsigned long) next_prev_parse_offset) - 13;
-      char *buf = new char[bufsize];
-      if (buf == NULL)
-        throw std::logic_error("DataUnitIO: Couldn't allocate memory for Data Unit");
-      stream.read(buf, bufsize);
-      d.strm.str(std::string(buf, bufsize));
-      delete[] buf;
-      break;
-    }
-    default:
-      throw std::logic_error("Stream Error: Nonconformant data unit type.");
-      break;
-    }
-  } else {
-    int bufsize = ((unsigned long) next_parse_offset) - 13;
-    char *buf = new char[bufsize];
-    if (buf == NULL)
-      throw std::logic_error("DataUnitIO: Couldn't allocate memory for Data Unit");
-    stream.read(buf, bufsize);
-    d.strm.str(std::string(buf, bufsize));
-    delete[] buf;
-  }
-
-  Bytes prefix(4);
-  stream >> prefix;
-
-  d.strm.copyfmt(stream);
-
+  throw std::logic_error("Stream assignment to DataUnit is no longer supported.");
```

Comment:
If this method is now defunct, it would be better to remove it entirely (which would produce a compile-time error if anyone tried to use it) rather than throw a runtime error (which we'd only discover 'in the field' given the lack of automated tests :)).